lib/authentication/view/edit_profile_page.dart

```diff
@@ -26,29 +26,55 @@ class _EditProfilePageState extends State<EditProfilePage> {
 
   final formKey = GlobalKey<FormState>();
 
-  AppDialog _deletionConfirmationDialog(BuildContext context) => AppDialog(
-        icon: Icon(Icons.warning, color: Colors.red),
-        title: S.of(context).actionDeleteAccount,
-        message: S.of(context).messageDeleteAccount +
-            ' ' +
-            S.of(context).messageCannotBeUndone,
-        actions: [
-          AppButton(
-            key: ValueKey('delete_account_button'),
-            text: S.of(context).actionDeleteAccount.toUpperCase(),
-            color: Colors.red,
-            width: 130,
-            onTap: () async {
+  AppDialog _deletionConfirmationDialog(BuildContext context) {
+    final passwordController = TextEditingController();
+    final passwordKey = GlobalKey<FormState>();
+    return AppDialog(
+      icon: Icon(Icons.warning, color: Colors.red),
+      title: S.of(context).actionDeleteAccount,
+      message: S.of(context).messageDeleteAccount +
+          ' ' +
+          S.of(context).messageCannotBeUndone,
+      content: [
+        Form(
```

Comment:
I wouldn't use form validation here. The toasts should be enough (check out the "Reset password" button on the login page).